File: js/escapeTime.js
load('./palette.js');
load('./image.js');

function EscapeTime(limits, size, maximumIterations) {
	this._minX = limits[0];
	this._maxX = limits[1];
	this._minY = limits[2];
	this._maxY = limits[3];
	this._imageWidth = size[0];
	this._imageHeight = size[1];
	this._maxIterations = maximumIterations;

	this.frame = null;
	this._image = null;
	this._colours = null;

	this.setUpImage = function(redLevel, blueLevel) {
		this._image = new Image(this._imageWidth, this._imageHeight);

		// Load the palette to find colours
		this._colours = new Palette(this._maxIterations, this._image, redLevel, blueLevel);
	}

	this.zoom = function(frame, factor) {
		this.frame = frame;
		this._minX /= factor;
		this._minY /= factor;
		this._maxX /= factor;
		this._maxY /= factor;
	}
}

File: lua/escapeTime.lua
require('palette')
require('image')

EscapeTime = {}
--EscapeTime.__index = EscapeTime

function EscapeTime:extend()
	local o = {}
	setmetatable(o, self)
	self.__index = self
	return o
end

function EscapeTime:new(limits, size, maximumIterations)
	local o = {}
	setmetatable(o, self)
	self.__index = self

	o._minX = limits[1]
	o._maxX = limits[2]
	o._minY = limits[3]
	o._maxY = limits[4]
	o._imageWidth = size[1]
	o._imageHeight = size[2]
	o._maxIterations = maximumIterations

	o.frame = nil
	o._image = nil
	o._colours = nil

	return o
end

function EscapeTime:setUpImage(redLevel, blueLevel)
	self._image = Image:new(self._imageWidth, self._imageHeight)

	-- Load the palette to find colours
	self._colours = Palette:new(self._maxIterations, self._image, redLevel, blueLevel)
end

function EscapeTime:zoom(frame, factor)
	self.frame = frame
	self._minX = self._minX / factor
	self._minY = self._minY / factor
	self._maxX = self._maxX / factor
	self._maxY = self._maxY / factor
end
